### DISCUSSION TITLE
mysqlclient 2.2.5 requires encrypted connection by default

### DISCUSSION THREAD

**Discussion by mpibpc-mroose** [PyMySQL/mysqlclient#740] [Issue report]
### Describe the bug

In encountered the problem described in https://stackoverflow.com/questions/79113919/ssl-is-required-but-the-server-does-not-support-it-django. Basically with the newest mysqlclient version TLS encrypted connections to databases are enforced. Even the proposed solution with setting `'ssl_mode': 'DISABLED'` does not work. When downgrading to 2.2.4 all is fine.

### Environment

MariaDB 10.6.7
Windows 11 23H2
Python 3.12
Django 5.1.2

### What version of mysqlclient do you use?

2.2.5

### Docker command to start MySQL server
Have a local version of MariaDB installed on a Windows machine (seem to be Windows related)

### Minimum but complete code to reproduce

```python
import MySQLdb
db=MySQLdb.connect(host="127.0.0.1", port=3306, user="foo", password="P@ssw0rd", database="some_sb")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\Local_Development\CellLineDatabase\venv-cell-line-db\Lib\site-packages\MySQLdb\__init__.py", line 121, in Connect
    return Connection(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Local_Development\CellLineDatabase\venv-cell-line-db\Lib\site-packages\MySQLdb\connections.py", line 195, in __init__
    super().__init__(*args, **kwargs2)
MySQLdb.OperationalError: (2026, 'TLS/SSL error: SSL is required, but the server does not support it')
```

### Schema and initial data required to reproduce.

```sql
CREATE USER 'foo'@'localhost' IDENTIFIED BY 'P@ssw0rd';
GRANT ALL PRIVILEGES ON some_sb.* TO 'foo'@'localhost' WITH GRANT OPTION;
CREATE DATABASE some_sb;
```

**Comment by methane** [comment]
duplicate of #729.
